### PR TITLE
Add tests to improve coverage.

### PR DESCRIPTION
**Pull request by bdice**
## Description
- Add more conditions to the test project to improve test coverage.
- Add tests of IgnoreConditions to improve coverage.

## Motivation and Context
These added tests should slightly improve coverage.

## Types of Changes
- [ ] Documentation update
- [ ] Bug fix
- [x] New feature
- [ ] Breaking change<sup>1</sup>

<sup>1</sup>The change breaks (or has the potential to break) existing functionality.

## Checklist:

- [x] I am familiar with the [**Contributing Guidelines**](https://github.com/glotzerlab/signac-flow/blob/master/CONTRIBUTING.md).
- [x] I agree with the terms of the [**Contributor Agreement**](https://github.com/glotzerlab/signac-flow/blob/master/ContributorAgreement.md).
- [x] My name is on the [list of contributors](https://github.com/glotzerlab/signac-flow/blob/master/contributors.yaml).
- [x] My code follows the [code style guideline](https://github.com/glotzerlab/signac-flow/blob/master/CONTRIBUTING.md#code-style) of this project.
- [x] The changes introduced by this pull request are covered by existing or newly introduced tests.

If necessary:
- [ ] I have updated the API documentation as part of the package doc-strings.
- [ ] I have created a separate pull request to update the [framework documentation](https://docs.signac.io/) on [signac-docs](https://github.com/glotzerlab/signac-docs) and linked it here.
- [ ] I have updated the [changelog](https://github.com/glotzerlab/signac-flow/blob/master/changelog.txt).

### CODE REVIEW

File: tests/test_project.py [hunk 1]
```diff
@@ -1492,7 +1492,7 @@ def test_run_with_operation_selection(self):
                 assert all(job.doc.get("test") for job in project)
                 project.run(names=["group2"])
                 assert all(job.isfile("world.txt") for job in even_jobs)
-                assert all(job.doc.get("test3") for job in project)
+                assert all(job.doc.get("test3_true") for job in project)
```
Comment (kidrahahjo) [inline tests/test_project.py:1495]:
```suggestion
                assert all(job.doc.get("test3_true") for job in project)
                assert all(not job.doc.get("test3_false") for job in project)
```

**Review by kidrahahjo — APPROVED** [review]
This looks good, just one more assert statement will be nice though.

**Review by vyasr — APPROVED** [review]
This also looks fine to me. Were `condition.[false|not_]` not tested anywhere before?

**Comment by bdice** [comment]
> This also looks fine to me. I'll let @bdice apply @kidrahahjo's fix in case you want to change anything before merging, but feel free to merge when you're ready. Were `condition.[false|not_]` not tested anywhere before?

Yes, I will apply that suggestion. You’re correct that these conditions were not tested before, I chose to do them because code coverage was missing.

**Pull request merged**